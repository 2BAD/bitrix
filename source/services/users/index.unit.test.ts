/* tslint-env jest */
// tslint:disable: no-expression-statement object-literal-sort-keys

import Users from '.'

const mockCall = jest.fn(() => Promise.resolve()) as any
const mockList = jest.fn(() => Promise.resolve()) as any
const users = Users({ call: mockCall })

const ID = '77'

describe('Users', () => {
  beforeEach(() => {
    mockCall.mockClear()
    mockList.mockClear()
  })

  it('should expose API methods', () => {
    expect(users).toMatchSnapshot()
  })

  describe('`get`', () => {
    it('should invoke `call`', async () => {
      await users.get(ID)
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })
})
