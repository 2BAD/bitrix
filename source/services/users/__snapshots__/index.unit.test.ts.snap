// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users \`get\` should invoke \`call\` 1`] = `
Array [
  Array [
    "user.get",
    Object {
      "id": "77",
    },
  ],
]
`;

exports[`Users should expose API methods 1`] = `
Object {
  "fields": [Function],
  "get": [Function],
}
`;
